java/core/src/main/java/co/worklytics/psoxy/storage/impl/CSVFileHandler.java

```diff
@@ -103,34 +119,56 @@ public byte[] handle(@NonNull InputStreamReader reader, @NonNull Sanitizer sanit
         }
 
 
+        List<String> columnNamesForOutputFile = Streams.concat(
+            applyReplacements(headers, columnsToRename).stream(),
+            columnsToDuplicate.values().stream())
+            .collect(Collectors.toList());
+
+
+        BiFunction<String, String, String> applyPseudonymizationIfAny = (outputColumnName, value) -> {
```

Comment:
this doesn't look like belongs here in the CSVFileHandler, but in a psedonymization utils / service.